### PR TITLE
[MI-2413]:Fixed merge conflicts and review comments on PR#89 

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
https://github.com/mattermost/mattermost-plugin-confluence/pulls/89

### CODE REVIEW

File: build/manifest/main.go [hunk 1]
```diff
@@ -103,19 +114,37 @@ func dumpPluginVersion(manifest *model.Manifest) {
...
 
 	if manifest.HasWebapp() {
+		// generate JSON representation of Manifest.
+		// JSON is very similar and compatible with JS's object literals. so, what we do here
+		// is actually JS code generation.
+		manifestBytes, err := json.MarshalIndent(manifest, "", "    ")
+		if err != nil {
+			return err
+		}
+		manifestStr := string(manifestBytes)
```
Comment (raghavaggarwal2308) [inline build/manifest/main.go:142]:
```suggestion

		manifestStr := string(manifestBytes)
```

File: server/kv.go [hunk 2]
```diff
@@ -656,13 +656,10 @@ func (store store) EnsureRSAKey() (rsaKey *rsa.PrivateKey, returnErr error) {
...
 		rsaKey = newRSAKey
 		store.plugin.debugf("Stored: RSA key")
-	}
 
-	// If we weren't able to save a new key above, another server must have beat us to it. Get the
-	// key from the database, and if that fails, error out.
-	if rsaKey == nil {
-		err = store.get(keyRSAKey, &rsaKey)
-		if err != nil {
+		// If we weren't able to save a new key above, another server must have beat us to it. Get the
```
Comment (raghavaggarwal2308) [inline server/kv.go]:
```suggestion
		// If we were not able to save a new key above, another server must have beaten us to it. Get the
```
maybe?

File: server/plugin.go [hunk 3]
```diff
@@ -318,7 +318,7 @@ func (p *Plugin) updateConfig(f func(conf *Config)) Config {
 }
 
 func (p *Plugin) GetPluginURLPath() string {
-	return "/plugins/" + manifest.ID + "/api/v1"
+	return "/plugins/" + manifest.Id + "/api/v1"
```
Comment (raghavaggarwal2308) [inline server/plugin.go:321]: why this change?

File: build/manifest/main.go [hunk 4]
```diff
@@ -103,19 +114,37 @@ func dumpPluginVersion(manifest *model.Manifest) {
...
 		); err != nil {
 			return errors.Wrap(err, "failed to write server/manifest.go")
 		}
 	}
 
 	if manifest.HasWebapp() {
+		// generate JSON representation of Manifest.
+		// JSON is very similar and compatible with JS's object literals. so, what we do here
+		// is actually JS code generation.
+		manifestBytes, err := json.MarshalIndent(manifest, "", "    ")
```
Comment (raghavaggarwal2308) [inline build/manifest/main.go:138]: Here we are using 4 space indent but above we are using 2. Is that intentional?

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: server/plugin.go (hunk 3 above)
Comment (Kshitij-Katiyar) [reply server/plugin.go:321]: Actually confluence master added some code in the build folder which was related to the creating of manifest file so due to that I copied the same code and used that to create the manifest file this change is because of that.

File: build/manifest/main.go (hunk 1 above)
Comment (Kshitij-Katiyar) [reply build/manifest/main.go:142]: copied this from the confluence master don't want this change to highlight there.

File: server/kv.go (hunk 2 above)
Comment (Kshitij-Katiyar) [reply server/kv.go]: both are same and have the same meaning as well.

File: server/kv.go [hunk 5]
```diff
@@ -16,14 +16,15 @@ import (
...
-	webhookKeyPrefix    = "webhook"
-	configKeyPrefix     = "_config"
+	keyInstances              = "instances/v3"
+	keyRSAKey                 = "rsa_key"
+	keyTokenSecret            = "token_secret"
+	prefixInstance            = "conf_instance_"
+	prefixUser                = "user_"
+	prefixOneTimeSecret       = "ots_" // + unique key that will be deleted after the first verification
+	webhookKeyPrefix          = "webhook"
+	configKeyPrefix           = "_config"
```
Comment (ayusht2810) [inline server/kv.go]: shouldn't it be like this `prefixWebhookKey` and `prefixConfigKey`?

File: build/manifest/main.go (hunk 4 above)
Comment (Kshitij-Katiyar) [reply build/manifest/main.go:138]: discussed

**Review by ayusht2810 — APPROVED** [review]

File: server/kv.go (hunk 5 above)
Comment (Kshitij-Katiyar) [reply server/kv.go]: they have followed these naming conventions everywhere that's why I have followed this

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/kv.go (hunk 5 above)
Comment (Kshitij-Katiyar) [reply server/kv.go]: fixed

**Review by manojmalik20 — APPROVED** [review]
